File: VB/03_Images/ConvertImageToPDF.vb
Imports Spire.Pdf
Imports Spire.Pdf.Graphics

Namespace ConvertImageToPDF
    Partial Public Class Form1
        Inherits Form
        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
            ' Create a new PdfDocument object
            Dim pdf As New PdfDocument()

            ' Add a section to the document
            Dim section As PdfSection = pdf.Sections.Add()

            ' Add a page to the document
            Dim page As PdfPageBase = pdf.Pages.Add()

            ' Load an image from file
            Dim image As PdfImage = PdfImage.FromFile("..\..\..\..\..\..\Data\bg.png")

            ' Calculate the fit rate for the image on the page
            Dim widthFitRate As Single = (image.PhysicalDimension.Width / page.Canvas.ClientSize.Width)
            Dim heightFitRate As Single = (image.PhysicalDimension.Height / page.Canvas.ClientSize.Height)
            Dim fitRate As Single = Math.Max(widthFitRate, heightFitRate)

            ' Calculate the dimensions of the image after fitting it on the page
            Dim fitWidth As Single = image.PhysicalDimension.Width / fitRate
            Dim fitHeight As Single = image.PhysicalDimension.Height / fitRate

            ' Draw the image on the page's canvas at position (0, 30) with the fitted dimensions
            page.Canvas.DrawImage(image, 0, 30, fitWidth, fitHeight)

            ' Specify the output file name for the converted PDF
            Dim output As String = "ConvertImageToPDF-result.pdf"

            ' Save the document to the specified file
            pdf.SaveToFile(output)

            ' Close the document
            pdf.Close()

            ' Launch the Pdf file
            PDFDocumentViewer(output)
        End Sub
        Private Sub PDFDocumentViewer(ByVal fileName As String)
            Try
                Process.Start(fileName)
            Catch
            End Try
        End Sub

    End Class
End Namespace

File: CS/03_Images/ConvertImageToPDF.cs
using System;
using System.Drawing;
using System.Windows.Forms;
using Spire.Pdf;
using Spire.Pdf.Graphics;

namespace ConvertImageToPDF

{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            // Create a pdf document with a section and page added.
            PdfDocument pdf = new PdfDocument();
            PdfSection section = pdf.Sections.Add();
            PdfPageBase page = pdf.Pages.Add();

            //Load a tiff image from system
            PdfImage image = PdfImage.FromFile(@"..\..\..\..\..\..\Data\bg.png");

            //Set image display location and size in PDF
            //Calculate rate
            float widthFitRate = image.PhysicalDimension.Width / page.Canvas.ClientSize.Width;
            float heightFitRate = image.PhysicalDimension.Height / page.Canvas.ClientSize.Height;
            float fitRate = Math.Max(widthFitRate, heightFitRate);

            //Calculate the size of image 
            float fitWidth = image.PhysicalDimension.Width / fitRate;
            float fitHeight = image.PhysicalDimension.Height / fitRate;

            //Draw image
            page.Canvas.DrawImage(image, 0, 30, fitWidth, fitHeight);

            //Save the result pdf 
            string output = "ConvertImageToPDF-result.pdf";
            pdf.SaveToFile(output);
            pdf.Close();

            //Launch the Pdf file
            PDFDocumentViewer(output);
        }
        private void PDFDocumentViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        } 

    }
}
